CRM/EventsExtras/Hook/BuildForm.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+use CRM_EventsExtras_ExtensionUtil as E;
+use CRM_EventsExtras_SettingsManager as SettingsManager;
+
+/**
+ * Abstract class for BuildForm Hook
+ */
+abstract class CRM_EventsExtras_Hook_BuildForm {
+
+  /**
+   * Event tab to display on the form 
+   */
+  protected $eventTab;
+
+  /**
+   * Constractor for BuildForm class
+   * 
+   * @param string $eventTab
+   * 
+   */
+  protected function __construct($eventTab){
+    $this->eventTab = $eventTab;
+    $templatePath = E::path() . '/templates/CRM/EventsExtras/Form/EventTabs.tpl';
+    CRM_Core_Region::instance('page-body')->add([
+      'template' => "{$templatePath}"
+    ]);
```

Comment:
Can we move this logic to add the template into another function e.g 'addEventTabTemplate', then we can call the function in the constructor. `$this->addEventTabTemplate()`